packages/babel-core/src/transformation/file/options/option-manager.js

```diff
@@ -260,6 +260,17 @@ export default class OptionManager {
       let presetLoc;
       if (typeof val === "string") {
         presetLoc = resolve(`babel-preset-${val}`, dirname) || resolve(val, dirname);
+
+        // trying to resolve @organization shortcat
+        // @foo/es2015 -> @foo/babel-preset-es2015
+        if (!presetLoc) {
+          let matches = val.match(/^(@[^/]+)\/([^/]+)(\/.+)?$/);
```

Comment:
By looking at the regex it seems that `@babel/react/optim`will be transformed to `@babel/babel-preset-react`, as the additional path is never checked or used. And I do not think this is intended?

I think you want to nest the last group within the group infront of it, as the additional path should still be part of the `presetPath`:
```
/^(@[^/]+)\/([^/]+(\/.+)?)$/
```

Can you confirm that, or correct me otherwise? :)